Core/Src/main.c

```diff
@@ -117,7 +148,147 @@ int main(void)
     MX_SPI2_Init();
     MX_TIM4_Init();
     /* USER CODE BEGIN 2 */
+    //select mode 0
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_12, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_10, GPIO_PIN_SET);
+
+    //write settings
+    HAL_SPI_Transmit(&hspi2, (uint8_t *)&ctrl_hum, 1, 100);
+    HAL_SPI_Transmit(&hspi2, (uint8_t *)&hum_osrs, 1, 100);
+    HAL_SPI_Transmit(&hspi2, (uint8_t *)&ctrl_meas, 1, 100);
+    HAL_SPI_Transmit(&hspi2, (uint8_t *)&meas_val, 1, 100);
+    HAL_SPI_Transmit(&hspi2, (uint8_t *)&config, 1, 100);
+    HAL_SPI_Transmit(&hspi2, (uint8_t *)&filter_off, 1, 100);
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_12, GPIO_PIN_SET);
+
+    spi_buff[0]=1;
+
+    //check status
+    while(spi_buff[0] & 0b00001000){ //bit masking need bit 3

```

Comment:
Here you could use bit shifting to make it more readable. 
If you need bit 3 then `while( spi_buf[0] & ( 1<< 3 ) )` would achieve the same result.
Or you could shift the first argument and compare it to 1. 
`while( (spi_buf[0] >> 3) & 1 )`